src/asm/stages/stage_2.asm

```diff
@@ -10,19 +10,31 @@ stage_2:
     lea si, second_stage_start_str
     call print_string_16
 
-    lea si, return_32_bit_mode_str
-    call print_string_16
+set_target_operating_mode:
+    # Some BIOSs assume the processor will only operate in Legacy Mode. We change the Target
+    # Operating Mode to "Long Mode Target Only", so the firmware expects each CPU to enter Long Mode
+    # once and then stay in it. This allows the firmware to enable mode-specifc optimizations.
+    # We save the flags, because CF is set if the callback is not supported (in which case, this is
+    # a NOP)
+    pushf
+    mov ax, 0xec00
+    mov bl, 0x2
+    int 0x15
+    popf
+
+
 
     call get_system_memory_map
 
-    call enter_protected_mode
-    push 0x8
-    lea eax, [stage_3]
-    push eax
-    retf
-    h:
-    	jmp h
+    lea si, return_32_bit_mode_str
+    call print_string_16
 
+    call enter_protected_mode 	# This function only sets the data segmentss
+    push 0x8					# Push the code segment in the gdt
```

Comment:
maybe set 0x8 to a const? this magic is unclear even with the comment

Reply:
no idea how to make consts in this assembly...

Reply:
actually i have no clue either, "GDT_CODE_SEGMANT equ 8" didn't work :(